src/browser/previewPDF.ts

```diff
@@ -0,0 +1,84 @@
+import puppeteer from 'puppeteer';
+import {
+  CHROMIUM_PATH,
+  margins,
+  pageHeight,
+  pageWidth,
+  setWindowProperty,
+} from './helpers';
+import ServiceNames from '../common/service-names';
+import templates from '../templates';
+import config from '../common/config';
+import renderTemplate, {
+  getHeaderAndFooterTemplates,
+} from '../server/render-template';
+
+const previewPdf = async (
+  url: string,
+  templateConfig: { service: ServiceNames; template: string },
+  templateData: Record<string, unknown>,
+  orientationOption?: boolean
+) => {
+  const browserMargins = {
```

Comment:
It looks like most of this code is also identical in the worker file. You could refactor them to use this as a setup function for puppeteer or something.